Limit lists to a certain number of entities

richard-cox opened the pull request:
- Allow a paginated action to define a maximum number of entities it should fetch. If this is exceeded do not depaginate response (fetch all pages in one go)
- In the list component show a custom warning message when there are too many results
- Allow user to reduce results by filtering server side
- Wire in above to app wall
- To test update GetAllApplications 'flattenPaginationMax' to 10
- fixes #2895

KlapTrap commented:
Bug:

1) Go to the app wall
2) Select a Cloud Foundry that has applications in the top filter panel
3) Select an Organisation that has applications
4) Select an Organisation that doesn't have any applications
5) Select the same Organisation from step 3, notice you don't get any applications in the list and are now in a broken state.

KlapTrap commented:
@richard-cox can you merge master into this PR? I'll review once that's done.

richard-cox commented:
@KlapTrap I've fixed points 1-3. Point 4 I've addressed (it now doesn't clear the list) however there's still a visible blip when changing a filter. This is due to the page$ observable returning an empty list.. due to the filter pipeline executing using the new filter values on the old page. I could fix this but it starts to get hacky (having something in the pagination object to block the page$ observable until cleared by a successful call?). Point 5 might be quite tricky to implement. I think at this stage, given how old this PR is and the amount of stacked PRs on top, we capture this as something to address separately.

richard-cox commented:
@KlapTrap Apart from the issue above, is this alright to merge? Just wary that you're away next week

KlapTrap commented:
Another issue :'(. On an empty list (in my example is was a application instance list) when you hit refresh the empty list message disappears while the list fetches.

I think we need some e2e tests covering these loading combinations (using current master as reference) as they're easy to break and easy to overlook.

KlapTrap commented:
If you're happy after fixing these issues, feel free to merge this in while i'm away to unblock all the things.

richard-cox commented:
After playing a fair amount of wackamole I've created #3305 to fix the remaining loading indicator issues. It needs a bit more of a rework and we need to unblock other issues. Once we've got those merged I'll come back and fix.
